File: part_1/hw4/ex_A_3_3_d.m
cvx_begin
    variable x;
    variable y;
    variable z1;
    variable z2;
    minimize(norm(x + y + 1));
    subject to
        z1 >= max(x,1);
        z2 >=  max(y,2)
        norm([z1, z2]) <= 3*x + y;
cvx_end

fprintf('status:'); 
disp(cvx_status);

fprintf('optimal value:'); 
disp(cvx_optval )

fprintf('optimal x:\n'); 
disp(x)

fprintf('optimal y:\n'); 
disp(y)

File: part_1/hw4/ex_A_3_3_d.py
import cvxpy as cp

# Create two scalar optimization variables.
x = cp.Variable()
y = cp.Variable()
z = cp.Variable(2)
z1 = cp.Variable()
z2 = cp.Variable()

# Create two constraints.
constraints = [z[0] >= cp.maximum(x, 1), z[1] >= cp.maximum(y, 2), 
               cp.norm(z) <= 3 * x + y]

# Form objective.
obj = cp.Minimize((x - y + 2)**2)

# Form and solve problem.
prob = cp.Problem(obj, constraints)
prob.solve()  # Returns the optimal value.
print("status:", prob.status)
print("optimal value", prob.value)
print("optimal var", x.value, y.value)

